components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/handler/mgm/StartRegistrationHandler.kt

```diff
@@ -189,6 +190,14 @@ internal class StartRegistrationHandler(
             "Empty member context in the registration request."
         }
 
+        when(val result = registrationContextCustomFieldsVerifier.verify(memberContext)) {
+            is RegistrationContextCustomFieldsVerifier.Result.Failure -> {
+                logger.info(result.reason)
+                throw InvalidRegistrationRequestException(result.reason)
+            }
+            RegistrationContextCustomFieldsVerifier.Result.Success -> {}
+        }
```

Comment:
Since nothing is done on the success branch could this be simplified to:
```suggestion
        if(registrationContextCustomFieldsVerifier.verify(memberContext) is RegistrationContextCustomFieldsVerifier.Result.Failure) {
                logger.info(result.reason)
                throw InvalidRegistrationRequestException(result.reason)
        }
```

Reply:
Or we could use the validateRegistrationRequest function like on line 189, which also logs a warn messages and throws InvalidRegistrationRequestException

Reply:
My (probably personal) opinion is we should be explicit here. If we added another failure type, say where we could retry (not really relevant here), we would want to handle it on this layer. I think we are using exceptions for error handling too much, this makes the code difficult to follow.

Reply:
I think the `if` version is still explicit, it's just focusing on the cases that it actually handles whereas this version introduces an empty code block for the success branch [which could be seen as a bit of a code smell](https://rules.sonarsource.com/css/RSPEC-4658)
(Happy to leave it as a when though if it's your preference)

Reply:
I changed this to use `validateRegistrationRequest`